plugins/locale.js

```diff
@@ -1,30 +1,33 @@
 import { join } from "node:path";
-import { existsSync, readFileSync } from "node:fs";
+import { exists } from "node:fs/promises";
 import fp from "fastify-plugin";
+import PathResolver from "../lib/path.js";
 import chalk from "chalk";
-import emojiFlags from "emoji-flags";
 
-// function that takes a locale string and returns the emoji flag for that locale
-// handles formats like en-US, en, en-US-POSIX
-// map countries to languages
-function getFlag(locale) {
-  if (!locale) return "";
-  const countryCode = locale.split("-")[1];
-  if (!countryCode) return "";
-  return `${emojiFlags.countryCode(countryCode).emoji}  `;
-}
+export default fp(async function locale(fastify, { cwd = process.cwd(), locale = "en", development }) {
+  const resolver = new PathResolver({ development, cwd });
 
-export default fp(async function locale(fastify, { cwd = process.cwd(), locale = "" }) {
-  const localeFilePath = join(cwd, "locale", locale) + ".json";
-  if (existsSync(localeFilePath)) {
-    try {
-      const translations = JSON.parse(readFileSync(localeFilePath, { encoding: "utf8" }));
-      fastify.decorate("translations", translations);
-      fastify.log.debug(
-        `${getFlag(locale)}${chalk.magenta("translations")}: loaded file "${localeFilePath}" for locale "${locale}"`
-      );
-    } catch (err) {
-      fastify.log.error(`Error reading translation file: ${localeFilePath}`, err);
+  const contentFile = await resolver.resolve("./content");
+  const compiledFileExtension = contentFile.typescript ? ".ts" : ".mjs";
+
+  const localeFilePath = join(cwd, "locales", locale) + compiledFileExtension;
+
+  fastify.decorate("readTranslations", async () => {
+    if (await exists(localeFilePath)) {
+      try {
+        const { messages } = await import(localeFilePath);
```

Comment:
You could probably move the exists check and the import to outside the decorator since they only ever happen 1x (import will cache the result). If you think it does need to happened repeatedly then you'll need to break the import cache with a timestamp like I've done in other places.